src/providers/pkcs11_provider/mod.rs

```diff
@@ -722,9 +727,24 @@ impl Provide for Pkcs11Provider {
                     error!("Error when extracting attribute: {}.", rv);
                     Err(ResponseStatus::PsaErrorCommunicationFailure)
                 } else {
+                    let mut modulus = attrs[0].get_bytes();
+                    let mut public_exponent = attrs[1].get_bytes();
+
+                    // To produce a valid ASN.1 RSAPublicKey structure, 0x00 is put in front of the positive
+                    // integer if highest significant bit is one, to differentiate it from a negative number.
+                    if modulus[0] & 0x80 == 0x80 {
```

Comment:
`if modulus[0] >= 128`
But I agree that the current check is more explicit as to what it's doing